controllers/discovery/searchUser.js

```diff
@@ -56,6 +56,10 @@ const SearchUser = async (req, res) => {
           u.combined_user_score,
           u.karma_score,
           u.is_karma_unlocked,
+          ARRAY( select name from topics as tp
+            left join user_topics as utp on tp.topic_id = utp.topic_id
+            where utp.user_id = u.user_id limit 3
+          ) as community_info,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The query to fetch `community_info` could potentially return an array with null values if there are no matching records in the `user_topics` table for a given `user_id`. This might not be the expected behavior on the frontend. Consider adding a condition to check if `utp.user_id` is not null before selecting the `name`.

```diff
-           ARRAY( select name from topics as tp
-             left join user_topics as utp on tp.topic_id = utp.topic_id
-             where utp.user_id = u.user_id limit 3
-           ) as community_info,
+           ARRAY( select name from topics as tp
+             left join user_topics as utp on tp.topic_id = utp.topic_id
+             where utp.user_id = u.user_id AND utp.user_id IS NOT NULL limit 3
+           ) as community_info,
```
Also, it's important to note that this subquery will only return up to 3 topics due to the `limit 3` clause. If more than 3 topics are associated with a user, only the first 3 (based on the natural order of the database) will be returned. If this is not the intended behavior, consider removing the limit or increasing it to match the requirements.


<!-- This is an auto-generated comment by OSS CodeRabbit -->